### ISSUE TITLE
Mailer Previews don't infer table_name for classes with a table_name_prefix

### ISSUE THREAD

**Issue by joshmn** [rails/rails#39194]
Interesting bug with Autoloader? I wasn't able to reproduce this with master and zeitwork, or master and classic.

### Steps to reproduce

See https://github.com/joshmn/mailer-preview-bug

Create a namespaced model, such as `rails g model Schools::Student name` where its `table_name` is prefixed from the module.

Create a mailer preview.

```ruby
module Schools
  class StudentMailerPreview < ActionMailer::Preview
    def welcome
      student = ::Schools::Student.first_or_create(name: "David")
      StudentMailer.with(student_id: student.id).welcome
    end
  end
end
```

Query is incorrect: `SQLite3::SQLException: no such table: students: SELECT "students".* FROM "students" ORDER BY "students"."id" ASC LIMIT ?`

If you explicitly specify `table_name` on the model, the query is executed correctly.

Note: Sometimes the `table_name` is inferred correctly. But after you hit an error, it'll stop resolving correctly. Sometimes it just doesn't work either.

### Expected behavior

The `table_name` should be prefixed correctly.

### Actual behavior

The `table_name` isn't prefixed correctly.

### System configuration
**Rails version**: 5.2.4.2

**Ruby version**: 2.5.6

**Comment by rafaelfranca** [comment]
Thank you so much for the bug report. This is a limitation of the classic reloader, that is now fixed on Rails 6.0 when using the zeitwerk reloader.

The reason for that is that the classic reloader will not load `schools.rb` because you previer is defining the module when you do `module Schools`. I think if you change your class definition to `class Schools::StudentMailerPreview < ActionMailer::Preview` that would work.

Closing this issue since Rails 5.2 is not supported anymore.

**Comment by joshmn** [comment]
Thanks @rafaelfranca, that's what I figured. Just thought I'd have this here for visibility in case someone else stumbles upon it :)
